prompt2model/utils/api_tools.py

```diff
@@ -40,13 +40,20 @@
 class APIAgent:
     """A class for accessing API-based models."""
 
-    def __init__(self, model_name: str = "gpt-3.5-turbo"):
+    def __init__(
+        self,
+        model_name: str = "gpt-3.5-turbo",
+        max_tokens: int | None = None,
```

Comment:
Should `max_tokens` be a property for the class initialization? Since all the other generation parameters are being passed to the completion function directly, maybe this one should be too?

Reply:
This is a good point, but it's an intentional design decision. Only properties that are exposed as agent parameters are easy for users to control, as most of the calls to generation are done internally to prompt2model and the users do not control them directly. One option would be to make them:

1. arguments of class initialization
2. arguments of the generation function, which are set by default to `None`

Then the generation function will use the class defaults unless they are overridden. I could do this in this PR or for another PR (I'm thinking maybe another PR as it becomes necessary because YAGNI).

Reply:
Right, that makes sense! Both options seem good for future contributions.